Automattic/wp-calypso
# Backup: Add option to mark sites as backup staging sites

## Proposed Changes

* Add new Jetpack Setting to mark sites as backup staging sites and make it available to sites with Jetpack Credentials through Jetpack Cloud.

## Screenshots

<img width="704" alt="image" src="https://user-images.githubusercontent.com/1488641/234578073-5b5bc5f0-f0a4-4ae4-80e6-6cf9b4f3e841.png">


## Demo

[screencast-jetpack.cloud.webm](https://user-images.githubusercontent.com/1488641/234577679-cd6ce363-a368-4f73-87ef-804f6c103616.webm)

## Testing Instructions

* Spin up a site with a Jetpack VaultPress Backup plan.
* Open a Jetpack Cloud live branch
* Select a site with a Jetpack VaultPress Backup plan.
* Navigate to `Settings` page
* If you don't have credentials configured, ensure the new `Set up as staging site` section is not visible.
* If you do, ensure the new section is visible.
* Play a bit, toggling the new option to ensure it is updating the staging flag. After refreshing the page, you should see the state you updated last time.

## Pre-merge Checklist
